ignore/src/types.rs

```diff
@@ -346,6 +346,16 @@ impl<'a> Glob<'a> {
     fn unmatched() -> Glob<'a> {
         Glob(GlobInner::UnmatchedIgnore)
     }
+
+    /// Return the file type defintion that matched, if any
```

Comment:
Could you fill out the docs a bit more here? Please also use a style consistent with the comments in the surrounding code (e.g., use punctuation).

I don't think you need much more here. But instead of saying, "if any," maybe we can say when a file type def is actually returned, e.g., if there was a match. When there's no match, then there's no available file type def.